DataDefinitions/Haulage.cs

```diff
@@ -69,6 +70,25 @@ public Haulage(long MissionId, string Name, string OriginSystem, int Amount, Dat
             remaining = Amount;
             expiry = Expiry;
             shared = Shared;
+
+            // Mechanism for identifying chained delivery and 'welcome' missions
+            typeEDName = Name.Split('_').ElementAtOrDefault(1)?.ToLowerInvariant();
+            switch (typeEDName)
```

Comment:
I understand, but I've given this some thought. This is truncated to handle haulage specific missions and must be in the ctor because object creation is not limited to the 'Mission accepted' event handler. For example, The 'Cargo depot' event handler also creates Haulage objects for shared missions.